src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle firstRectangle = (Rectangle) rectangle;
+            return ((width == null) || width.equals(firstRectangle.width))
+                    && (length == null || length.equals(firstRectangle.length))
+                    && (color == null || color.equals(firstRectangle.color));
```

Comment:
what if width of current object  == null, but firstRectangle.width has some other value? Should the result be true?